docker-app/qfieldcloud/core/serializers.py

```diff
@@ -535,3 +539,44 @@ class LatestPackageSerializer(serializers.Serializer):
     package_id = serializers.UUIDField()
     packaged_at = serializers.DateTimeField()
     data_last_updated_at = serializers.DateTimeField()
+
+
+class TeamSerializer(serializers.ModelSerializer):
+    organization = serializers.StringRelatedField(source="team_organization")
+
+    class Meta:
+        model = Team
+        fields = ("username", "organization")
+
+
+class TeamMemberSerializer(serializers.ModelSerializer):
+    member = serializers.CharField()
+
+    def to_internal_value(self, data):
+        validated_data = super().to_internal_value(data)
+        email_or_username = data.get("member")
+        existing_username = self.context["view"].kwargs.get("member_username")
+
+        if existing_username is not None:
+            if not email_or_username:
+                raise serializers.ValidationError(
+                    {"member": "Username must be provided"}
+                )
+
+            try:
+                if User.objects.fast_search(email_or_username) > 0:
+                    raise serializers.ValidationError(
+                        {
+                            "member": f"A user with username '{email_or_username}' already exists."
+                        }
+                    )
+            except User.DoesNotExist:
+                email_or_username = self.context["view"].kwargs.get("member_username")
```

Comment:
```suggestion
```

We might only need the check for uniqueness.

Reply:
Check the final comment

Reply:
See my final answer https://github.com/opengisch/QFieldCloud/pull/1031#discussion_r1868518703. Closing this thread.